packages/core/src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -137,16 +161,25 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
 
     private async _setupPayment(
         braintreePaypalCheckout: BraintreePaypalCheckout,
-        shippingAddress?: Address | null,
-        onError?: (error: BraintreeError | StandardError) => void
+        braintreepaypal: BraintreePaypalButtonInitializeOptions
     ): Promise<string> {
-        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        const { buyNowInitializeOptions, shippingAddress, onPaymentError } = braintreepaypal;
+        let state: InternalCheckoutSelectors;
 
         try {
-            const checkout = state.checkout.getCheckoutOrThrow();
-            const config = state.config.getStoreConfigOrThrow();
+            this._buyNowCart = await this._createBuyNowCart({ buyNowInitializeOptions });
+
+            if (this._buyNowCart) {
+                state = this._store.getState();
+            } else {
+                state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
```

Comment:
Do we still need to load default checkout here?

Reply:
Yes
This is required because we need get checkout for the non 'Buy Now' logic

Reply:
But we are getting it in the initialization method. Why should we repeat the loading here?

Reply:
@serhii-tkachenko We can create a new private variable for **state** like `private _state: InternalCheckoutSelectors` and get it from there. How do you think?

Reply:
It is not necessary due to `this._store.getState()` method, because the state is already has all data that we need. I think that it is a question of another PR, lets skip it for now